lib/pal/InformationProviderImpl.cpp

```diff
@@ -73,11 +73,11 @@ namespace PAL_NS_BEGIN {
             // be deleted.  The current design is that IPropertyChangedCallback is
             // not refcount'ed.  Should we refcount it?
 
+            std::lock_guard<std::mutex> lock(m_lock);
```

Comment:
While this rightly increases the scope of mutex lock, it looks like it's not required to keep lock explicitly during callback function call (`callback::OnChanged`)

Reply:
Thanks for the comments! Any suggestions to this? Can we unlock?

Reply:
Don't see any straightforward fix. Either 
 - move the`local_callbacks` vector definition outside the lock guard. or
 - Use explicit lock/unlock instead of lock_guard.

Reply:
In general fine with it. Bringing `local_callbacks` vector outside may have some memory overhead in terms of heap allocation (std::allocator) every time the `InformatonProviderImpl::OnChanged` is called. But this should be negligible as most of the STL implementations have initial reserve capacity as `0` for std::vector. Still would be good if we can have another eye on this.

Reply:
We can wait for a couple of days for comments and merge.